tests/end2end/features/test_history_bdd.py

```diff
@@ -17,36 +17,30 @@
 # You should have received a copy of the GNU General Public License
 # along with qutebrowser.  If not, see <http://www.gnu.org/licenses/>.
 
-import os.path
+import logging
+import re
 
 import pytest_bdd as bdd
-bdd.scenarios('history.feature')
 
+bdd.scenarios('history.feature')
 
-@bdd.then(bdd.parsers.parse("the history file should contain:\n{expected}"))
-def check_history(quteproc, httpbin, expected):
-    history_file = os.path.join(quteproc.basedir, 'data', 'history')
-    quteproc.send_cmd(':save history')
-    quteproc.wait_for(message=':save saved history')
 
-    expected = expected.replace('(port)', str(httpbin.port)).splitlines()
+@bdd.then(bdd.parsers.parse("the history should contain:\n{expected}"))
+def check_history(quteproc, httpbin, tmpdir, expected):
+    path = tmpdir / 'history'
+    quteproc.send_cmd(':debug-dump-history "{}"'.format(path))
+    quteproc.wait_for(category='message', loglevel=logging.INFO,
+                      message='Dumped history to {}'.format(path))
 
-    with open(history_file, 'r', encoding='utf-8') as f:
-        lines = []
-        for line in f:
-            if not line.strip():
-                continue
-            print('history line: ' + line)
-            atime, line = line.split(' ', maxsplit=1)
-            line = line.rstrip()
-            if '-' in atime:
-                flags = atime.split('-')[1]
-                line = '{} {}'.format(flags, line)
-            lines.append(line)
+    with open(path, 'r', encoding='utf-8') as f:
+        # ignore access times, they will differ in each run
+        actual = '\n'.join(re.sub('^\\d+-?', '', line).strip()
+                           for line in f.read().splitlines())
```

Comment:
Just saw a20aaab895eb5cce57c214999eb3e90e506867e0 - what I meant earlier is `for line in f` without `.read()`, i.e. iterating over the file (not the string).